IBPSA/Fluid/Movers/BaseClasses/Euler/getPeak.mo

```diff
@@ -0,0 +1,225 @@
+        dp:= pressure.dp;
+        for i in 1:n loop
+          P[i]:= IBPSA.Utilities.Math.Functions.smoothInterpolation(
+                   x=V_flow[i],
+                   xSup=power.V_flow,
+                   ySup=power.P,
+                   ensureMonotonicity=false);
+        end for;
+      else
+        V_flow:= power.V_flow;
+        P:= power.P;
+        for i in 1:n loop
+          dp[i]:= IBPSA.Utilities.Math.Functions.smoothInterpolation(
+                    x=V_flow[i],
+                    xSup=pressure.V_flow,
+                    ySup=pressure.dp,
+                    ensureMonotonicity=true);
+        end for;
+      end if;
+    end if;
+
+    // Compute efficiency array
+    eta:=V_flow.*dp./P;
+    etaLes:=n<4;
+    etaMon:=IBPSA.Utilities.Math.Functions.isMonotonic(
+      x=eta,
+      strict=false);
+
+    if etaLes or etaMon then
+    // If less than four data points are provided or the efficiency curve is
+    //   monotonic, use the one of two which has the higher efficiency:
+    //   1. the interpolated point at half of max flow;
+    //   2. the available point with the highest efficiency.
+      peak.eta:=IBPSA.Utilities.Math.Functions.smoothInterpolation(
+               x=V_flow_hal,
+               xSup=V_flow,
+               ySup=eta,
+               ensureMonotonicity=
+                 IBPSA.Utilities.Math.Functions.isMonotonic(eta, strict=false));
+      if peak.eta>max(eta) then
+             peak.V_flow:=V_flow_hal;
+             peak.dp:=dpHalFlo;
+      else
+        peak.eta:=max(eta);
+        for i in 1:n loop
+          if abs(eta[i]-peak.eta)<1E-6 then
+            peak.V_flow:=V_flow[i];
+            peak.dp:=dp[i];
+          end if;
+        end for;
+      end if;
+    else
+
+      // Constructs the matrix equation A b = y for quartic regression, where
+      //   A is a 5-by-n design matrix,
+      //   b is a parameter vector with length 5,
+      //   and y = eta is the response vector with length n.
+      A[:,1]:=ones(n); // Avoids 0^0
+      for i in 2:5 loop
+        A[:,i]:=V_flow[:].^(i-1);
+      end for;
+      b:=Modelica.Math.Matrices.leastSquares(A,eta);
+
+      // Solve the derivative for the max eta and corresponding V_flow.
+      // This assumes that there will only be one real solution
+      //   and it falls within the range of V_flow[:].
+      r:=Modelica.Math.Polynomials.roots({b[5]*4,b[4]*3,b[3]*2,b[2]});
+      for i in 1:3 loop
+        if abs(r[i,2])<=1E-6 and
+          r[i,1]>V_flow[1] and r[i,1]<V_flow[n] then
+          peak.V_flow:=r[i,1];
+        end if;
+      end for;
+      peak.eta:=IBPSA.Utilities.Math.Functions.smoothInterpolation(
+        x=peak.V_flow,
+        xSup=V_flow,
+        ySup=eta,
+        ensureMonotonicity=false);
+      peak.dp:=IBPSA.Utilities.Math.Functions.smoothInterpolation(
+        x=peak.V_flow,
+        xSup=V_flow,
+        ySup=dp,
+        ensureMonotonicity=false);
+    end if;
+  end if;
+
+  annotation(smoothOrder=1,
```

Comment:
The annotation `smoothOrder=1` must be removed as this function is not C1.

Reply:
Corrected in 02bdb25.